server/src/main/java/io/druid/server/coordinator/CostBalancerStrategy.java

```diff
@@ -268,5 +277,28 @@ protected double computeCost(
     return bestServer;
   }
 
+  @Override
+  public boolean steady(final List<ServerHolder> serverHolders)
+  {
+    if (balancerSlope <= BalancerStrategy.DEFAULT_BALANCER_SLOPE) {
+      return false;
+    }
+    double totalCost = 0;
+    double maxCost = Double.MIN_VALUE;
+    double minCost = Double.MAX_VALUE;
+    for (ServerHolder server : serverHolders) {
+      double serverCost = calculateServerCost(server);
+      totalCost += serverCost;
+      maxCost = Math.max(maxCost, serverCost);
+      minCost = Math.min(minCost, serverCost);
+    }
+    double average = totalCost / serverHolders.size();
+    double floor =  average * (1 - balancerSlope);
+    double ceiling =  average * (1 + balancerSlope);
+    if (maxCost < ceiling && minCost > floor) {
+      return true;
+    }
+    return false;
+  }
```

Comment:
ok, i guess separate tiers are tried separately so that wouldn't be an issue. however, same issue will occur if one tier contained historicals of different capacities.